backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h

```diff
@@ -324,36 +324,39 @@ template <typename Torus> struct int_radix_lut {
 
   bool gpu_memory_allocated;
 
-  cudaEvent_t event_scatter_in;
-  cudaEvent_t *event_scatter_out;
-  cudaEvent_t event_broadcast;
-
-  int_radix_lut(CudaStreams streams, int_radix_params params, uint32_t num_luts,
-                uint32_t num_radix_blocks, bool allocate_gpu_memory,
-                uint64_t &size_tracker) {
+  CudaStreamsWorkersWaitFirstBarrier multi_gpu_scatter_barrier,
+      multi_gpu_broadcast_barrier;
+  CudaStreamsFirstWaitsWorkersBarrier multi_gpu_gather_barrier;
 
+  void setup_config_and_degrees(CudaStreams streams, int_radix_params params,
```

Comment:
Not sure we should have this function: or we could only have one to allocate degrees on the CPU, named `allocate_degrees_on_cpu` maybe.